src/main/java/org/opentripplanner/graph_builder/module/DirectTransferGenerator.java

```diff
@@ -116,6 +118,15 @@ public void buildGraph() {
             if (sd.stop == stop) {
               continue;
             }
+            var toParentStation = sd.stop.getParentStation();
+            if (OTPFeature.NoTransfersOnIsolatedStops.isOn()) {
+              if (
+                (toParentStation != null && toParentStation.isNoTransfers()) ||
+                (fromParentStation != null && fromParentStation.isNoTransfers())
+              ) {
+                continue;
+              }
```

Comment:
Instead of implementing the business logic for if a transfer is allowed or not here it should be pushed into the stop class. 
```suggestion
              if (sd.stop.transfersNotAllowed()) {
                continue;
              }
```
The `transfersNotAllowed` should have JavaDoc documenting under witch condition that can happen.